include/camp/defines.hpp

```diff
@@ -15,7 +15,7 @@ For details about use and distribution, please read LICENSE and NOTICE from
 #include <cstdint>
 #include <string>
 
-#if defined(__CUDACC__)
+#ifdef CAMP_HAVE_CUDA
```

Comment:
I'm not sure this is quite the right solution longer term since it only includes the header this is being read into a translation unit being compiled by a cuda compiler.  It's probably the right thing to do right now though, since we don't have another way to configure that.

Anyway, I think this is fine, but either this needs to move or the definition does, since the macro gets defined on line 74.